common/postgresql-ng/templates/deployment.yaml

```diff
@@ -53,7 +53,7 @@ spec:
         - name: DEPLOYMENT_NAME
           value: {{ $deployment_name }}
         - name: USERS
-          value: {{ keys (.Values.users | required ".Values.users must be configured") | sortAlpha | join " " | quote }}
+          value: "{{ range $user, $settings := (.Values.users | required ".Values.users must be configured") }}{{ if and (typeIs "map[string]interface {}" $settings) (hasKey $settings "secretName") }}{{ $user }} {{ end }}{{ end }}"
```

Comment:
This will be unstable across different template runs and produce noise in the diffs. We should still iterate ove `keys (...) | sortAlpha`. I suggest moving that entire expression into a separate variable to improve readability.

Also, the template function `and` does not short-circuit like `&&` does, so this should use layered conditions to be safe. All in all:

```
{{- $allUserNames := keys (.Values.users | required ".Values.users must be configured") | sortAlpha }}
value: "{{ range $user := $allUserNames }}{{ $settings := index $.Values.users $user }}{{ if typeIs "map[string]interface {}" $settings }}{{ if hasKey $settings "secretName" }}{{ $user }} {{ end }}{{ end }}{{ end }}"
```

Yes, I realize that this is not very nice.